pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -457,8 +460,30 @@ func (pc *PodController) configurePodSecondaryNetwork(pod *corev1.Pod, networkLi
 			}
 		}
 
+		// Generate a new MAC address, if the secondary interface MAC was not provided in the
+		// Pod annotation.
+		var mac net.HardwareAddr
```

Comment:
it may be a bit cleaner / readable to define a closure for this:

```go
getRequestedMac := func() (net.HardwareAddr, error) {
        // put all the logic here, the call to net.ParseMAC will be simpler
}

mac, err := getRequestedMac()
if err != nil {
        klog.ErrorS(...)
        // Do not return error: no need to requeue.
        continue
}
```